### DISCUSSION TITLE
Observation about refactoring pocketbase.go and main.go

### DISCUSSION THREAD

**Discussion by dschissler** [pocketbase/pocketbase#5677] [Q&A]
@ganigeorgiev If you moved the command installations from `Start` in `pocketbase.go` to `main.go` then it would be a much more perfectly generalized starting point for some more involved custom integration of Pocketbase.

I don't see the point of having `Start` command setup in `pocketbase.go` because very likely anyone who is wanting to create a new `main()` isn't going to use `Start` anyways.

With a bit of polishing and cleanup then PB becomes the obvious choice for creating [charmbracelet/wish](https://github.com/charmbracelet/wish) SSH apps using PB db setup without the web API (once it's been setup there), as db aware system maintenance utilities (using the built in cron), and many other things.

I already know that pb as just a db works perfectly well because I'm using it with the underlying [gliderlabs/ssh](https://github.com/gliderlabs/ssh) tech that Wish uses for bubble tea programs over SSH.  I think that it would be good if you just nudged the code more in that direction and made it clear that it's capable of doing it.  Then the community would take it from there.

----

Also, could `hideStartBanner` be just a command level flag for serve command instead of it being a `pocketbase.go` config option?

```golang
// Start starts the application, aka. registers the default system
// commands (serve, migrate, version) and executes pb.RootCmd.
func (pb *PocketBase) Start() error {
	// register system commands
	pb.RootCmd.AddCommand(cmd.NewSuperuserCommand(pb))
	pb.RootCmd.AddCommand(cmd.NewServeCommand(pb, !pb.hideStartBanner))

	return pb.Execute()
}
```

**Comment by ganigeorgiev** [comment]
@dschissler Please don't ping me. I've unsubscribed from the discussions notifications for a reason (_https://github.com/pocketbase/pocketbase/discussions/5555_).

I don't understand what do you mean by _"I don't see the point of having Start command"._

`Start()` does exactly what the name implies - it "starts" the application.

You don't have to use it if you have a custom startup operation and/or don't want for example the default commands to be registered. In fact you can even create your entire custom app initialization wrapper by instantiating manually `core.BaseApp`.

Regarding `hideStartBanner` - it is just a [`pocketbase.Config`](https://pkg.go.dev/github.com/pocketbase/pocketbase#Config) option. There is no binded command line flag for it.

I don't exactly understand your use case but I don't think it is a standard one. Feel free to elaborate how exactly you imagine the PocketBase startup procedures and public APIs should look like and I may consider it but there are no promises that I will follow and implement your demands.

**Reply by dschissler** [reply]
`main.go` becomes the place where the two commands are registered (using the same api as migratecmd).  Put the banner setup as a flag in servecmd with perhaps and overridable default by a servercmd.Config option.

`main.go`
```golang
func main() {
	migratecmd.MustRegister(app, app.RootCmd, migratecmd.Config{
		TemplateLang: migratecmd.TemplateLangJS,
		Automigrate:  automigrate,
		Dir:          migrationsDir,
	})
	superusercmd.MustRegister(app, app.RootCmd)
	servecmd.MustRegister(app, app.RootCmd)

	if err := app.Execute(); err != nil {
		log.Fatal(err)
	}
```

Delete `pocketbase.go` Start, and possibly rename Execute to Start.

**Reply by dschissler** [reply]
> You don't have to use it if you have a custom startup operation and/or don't want for example the default commands to be registered. In fact you can even create your entire custom app initialization wrapper by instantiating manually core.BaseApp.

I thought about that, but `pocketbase.go` is a great starting point already that is already really well generalized (except for a few unnecessary things) while also handling a lot of the hard and weird stuff like signal handling.  This is why I'm saying to make `pocketbase.go` perfectly generalized and to put all of the command setup into `main.go`.

`pocketbase.go` is a whole lot more generalized now than when I first starting using it, but it could still be cleaned up a bit so that it perfectly fits all universally applicable `core.Base` settings.
